iidm/iidm-util/src/main/java/com/powsybl/iidm/network/util/TerminalChooser.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.util;
+
+import com.powsybl.iidm.network.ConnectableType;
+import com.powsybl.iidm.network.Injection;
+import com.powsybl.iidm.network.Terminal;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * A class which allows to choose a terminal, according to given filters, among several terminals
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public final class TerminalChooser {
+
+    private final List<Predicate<Terminal>> rules;
+
+    public static TerminalChooser getDefaultSlackTerminalChooser() {
+        List<Predicate<Terminal>> rules = new ArrayList<>();
+        rules.add(t -> t.getConnectable().getType() == ConnectableType.BUSBAR_SECTION);
+        rules.add(t -> t.getConnectable() instanceof Injection);
```

Comment:
They could also be added if you think it makes sense, but @geofjamg would only leave the Busbar predicate... so that's a compromise? Anyway that's only the default chooser, other choosers can be defined and used.

Reply:
There is a risk that there is no busbar section (Bus/Breaker VoltageLevel) and no injection neither. For instance, if we consider a star-like network, the star bus is a good slack bus, even there is no injection connected to it.

I would add the 3 other predicates to be sure you can always attach the extension to a terminal.

Can you please add a unit test using the Eurostag Tutorial 1, and ensure your are able to set the slack terminal